client/src/main/scala/com.crobox.clickhouse/internal/ClickHouseExecutor.scala

```diff
@@ -24,9 +28,58 @@ private[clickhouse] trait ClickHouseExecutor extends LazyLogging {
   protected val hostBalancer: HostBalancer
   protected def config: Config
 
-  private lazy val pool = Http().superPool[Promise[HttpResponse]]()
+  lazy val (progressQueue, progressSource) = {
+    val builtSource = Source
+      .queue[String](1000, OverflowStrategy.dropHead)
+      .map(queryAndProgress => {
+        queryAndProgress.split("\n", 2).toList match {
+          case queryId :: progressJson :: Nil =>
+            Try {
+              val parsedJson = JSON.parseFull(progressJson).map(_.asInstanceOf[Map[String, String]])
+              if (parsedJson.isEmpty || parsedJson.get.size != 3) {
+                throw new IllegalAccessException(s"Cannot extract progress from $parsedJson")
+              } else {
+                val jsonMap = parsedJson.get
+                ClickhouseQueryProgress(
+                  queryId,
+                  Progress(jsonMap("read_rows").toLong, jsonMap("read_bytes").toLong, jsonMap("total_rows").toLong)
+                )
+              }
+            } match {
+              case Success(value) => Some(value)
+              case Failure(exception) =>
+                logger.warn(s"Failed to parse json $progressJson", exception)
+                None
+            }
+          case other @ _ =>
+            logger.warn(s"Could not get progress from $other")
+            None
+
+        }
+      })
+      .collect {
+        case Some(progress) => progress
+      }
+      .withAttributes(ActorAttributes.supervisionStrategy({
+        case ex @ _ =>
+          logger.warn("Failed in the progress streaming", ex)
```

Comment:
Wierd english, perhaps use `Detected failure in the query progress stream, resuming operation.`